sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,35 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
+   */
+  object SubstituteHint extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case logical: LogicalPlan => logical transformDown {
+        case h @ Hint(name, parameters, child)
+            if Seq("BROADCAST", "BROADCASTJOIN", "MAPJOIN").contains(name.toUpperCase) =>
+          var resolvedChild = child
+          for (table <- parameters) {
+            var stop = false
+            resolvedChild = child.transformDown {
+              case r @ BroadcastHint(UnresolvedRelation(_, _)) => r
+              case r @ UnresolvedRelation(t, _) if !stop && t.table == table =>
+                stop = true
+                BroadcastHint(r)
+            }
```

Comment:
Hint are a kind of comment. All the exceptional cases are ignored silently. That is the basic policy.